tests/FSharp.Test.Utilities/TestFramework.fs

```diff
@@ -10,28 +10,59 @@ open Scripting
 open Xunit
 open FSharp.Compiler.IO
 
-let inline getTestsDirectory src dir = src ++ dir
-
-// Temporary directory is TempPath + "/FSharp.Test.Utilities/" date ("yyy-MM-dd")
-// Throws exception if it Fails
-let tryCreateTemporaryDirectory () =
-    let date() = DateTime.Now.ToString("yyyy-MM-dd")
-    let now() = $"{date()}-{Guid.NewGuid().ToString()}"
-    let directory = Path.Combine(Path.GetTempPath(), now()).Replace('-', '_')
-    Directory.CreateDirectory(directory).FullName
-
-// Create a temporaryFileName -- newGuid is random --- there is no point validating the file already exists because: threading and Path.ChangeExtension() is commonly used after this API
-let tryCreateTemporaryFileName () =
-    let directory = tryCreateTemporaryDirectory ()
-    let fileName = ("Temp-" + Guid.NewGuid().ToString() + ".tmp").Replace('-', '_')
-    let filePath = Path.Combine(directory, fileName)
-    filePath
-
-// Create a temporaryFileName -- newGuid is random --- there is no point validating the file already exists because: threading and Path.ChangeExtension() is commonly used after this API
-let tryCreateTemporaryFileNameInDirectory (directory: DirectoryInfo) =
-    let fileName = ("Temp-" + Guid.NewGuid().ToString() + ".tmp").Replace('-', '_')
-    let filePath = Path.Combine(directory.FullName, fileName)
-    filePath
+let getShortId() = Guid.NewGuid().ToString().[..7]
+
+// Temporary directory is TempPath + "/FSharp.Test.Utilities/yyy-MM-dd-xxxxxxx/"
+let tempDirectoryOfThisTestRun =
+    let tempDir = Path.GetTempPath()
+    let today = DateTime.Now.ToString("yyyy-MM-dd")
+    DirectoryInfo(tempDir)
+        .CreateSubdirectory($"FSharp.Test.Utilities/{today}-{getShortId()}")
+        .FullName
+
+let createTemporaryDirectory (part: string) =
+    DirectoryInfo(tempDirectoryOfThisTestRun)
+        .CreateSubdirectory($"{part}-{getShortId()}")
+        .FullName
+
+let getTemporaryFileName () =
+    (createTemporaryDirectory "temp") ++ $"tmp_{getShortId()}"
+
+let getTemporaryFileNameInDirectory (directory: string) =
+    directory ++ $"tmp_{getShortId()}"
+
+// Well, this function is AI generated.
```

Comment:
This on its own sounds scary, but looks like this is an adoption of this [docs](https://learn.microsoft.com/en-us/dotnet/standard/io/how-to-copy-directories), maybe include the link :D